src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (compareFunction === undefined) {
+      for (let i = 0; i < this.length - 1; i++) {
+        if (this[i] + '' > this[i + 1] + '') {
+          const currentIndex = this[i];
+
+          this[i] = this[i + 1];
+          this[i + 1] = currentIndex;
+
+          i = -1;
+        }
+      }
+    } else {
+      for (let i = 0; i < this.length - 1; i++) {
+        if (compareFunction(this[i], this[i + 1]) > 0) {
+          const currentIndex = this[i];
+
+          this[i] = this[i + 1];
+          this[i + 1] = currentIndex;
+
+          i = -1;
+        }
```

Comment:
The same inefficiency issue is present here as well. After swapping elements, you should continue checking from the next position rather than starting over.